Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -112,6 +135,12 @@ bool OverlayPanel::eventFilter(QObject* watched, QEvent* event) {
   } else if(watched == parentWidget()->window()) {
     if(event->type() == QEvent::Move) {
       position();
+    } else if(event->type() == QEvent::WindowDeactivate &&
+        m_is_closed_on_blur && !isActiveWindow() && m_was_activated) {
```

Comment:
When does the situation of '!isActiveWindow() && m_was_activated' happen?

Reply:
This event is when the parent widget's window is deactivated, so this covers the cases where:

1. This instance of the OverlayPanel was activated. In this case m_was_activated will be true, but since this is the active window it shouldn't close.
2. Some unrelated window (or a sibling) was activated, but since this instance of the OverlayPanel was never activated, then it shouldn't close.

There's typically no situation where the parent widget's window is deactivated and this instance of the OverlayPanel was already activated previously, but since it's shorter to include both statements in the same condition and they don't interfere with each other I left it that way. I say typically because we don't currently change m_is_closed_on_blur at runtime, but a requirement could come up in the future where that happens and this condition would still satisfy the spec, which could be an argument for keeping it this way.

Reply:
I think that m_was_activated and isActiveWindow() should be equivalent, so m_was_activated and !isActiveWindow() always are opposite.

Reply:
The difference is that m_was_activated is never updated after it's set to true, but isActiveWindow() is conditional on the state of the window. m_was_activated is false initially, but as soon as the OverlayPanel is activated m_was_activated is set to true and it stays true for the lifetime of the OverlayPanel.

Reply:
I see. Thank you for your clarification.